p2p/security/noise/session_transport.go

```diff
@@ -31,9 +31,10 @@ type EarlyDataHandler interface {
 	Received(context.Context, net.Conn, []byte) error
 }
 
-func EarlyData(h EarlyDataHandler) SessionOption {
+func EarlyData(initiator, responder EarlyDataHandler) SessionOption {
```

Comment:
We should have a comment here explaining when the initiator and responder sends and receives this early data. "Early data" isn't a noise concept, we made this up. We should be clear that early data sent by the initiator is sent **plaintext** in the **first** message. And the responder early data is sent in the second message. 

I got tripped up by this recently since in the js-libp2p noise implementation early data from the initiator is sent in the **third** message so that it's encrypted.

I think a comment with at least a big warning about the plaintext part and an explanation of when the data is sent would help here.

Reply:
oh and update the comments on EarlyDataHandler: https://github.com/libp2p/go-libp2p/blob/4f8a466b4cf3da8df34ea2cf26b4045aa435f2bb/p2p/security/noise/session_transport.go#L25

Reply:
@marten-seemann I made these changes in https://github.com/libp2p/go-libp2p/pull/1750/commits/0fe5519d217b2a404a6dac55fe5ce7c8b0fbeb06 to help. Feel free to remove/revert that commit if you want to do something different.

Reply:
@MarcoPolo Thank you for updating the documentation. I really like your change!